include/utils/kahan.hpp

```diff
@@ -43,4 +45,94 @@ T kahan(InputIt first, InputIt last, T init) {
     return sum;
 }
 
+template<class InputIt, class T>
+T kahanBabushka(InputIt first, InputIt last, T init) {
+    T sum = init;
+    T cs = 0.;
+    T ccs = 0.;
+    T c = 0.;
+    T cc = 0.;
+
+    for (auto input = first; input != last; input++) {
+        T t = sum + *input;
+        if(std::abs(sum) >= std::abs(*input))
+            c = (sum - t) + *input;
+        else
+            c = (*input - t) + sum;
+        sum = t;
+        t = cs + c;
+        if(std::abs(cs) >= std::abs(c))
+            cc = (cs - t) + c;
+        else
+            cc = (c - t) + cs;
+        cs = t;
+        ccs = ccs + cc;
+    }
+    return sum + cs + ccs;
+}
+
+std::tuple<double,double,int,int> kahanBabushkaTuple(std::vector<BankedParticle>::iterator first, std::vector<BankedParticle>::iterator last) {
```

Comment:
If we change the name of the function above to just `kahan`, we should maybe call this `kahan_bank` or something similar, so we know it only works on a vector of `BankedParticle`. In a comment above, you should also indicate what each entry of the tuple is holding.